### PR TITLE
feat(web): suggest people when typing a name

### PR DESCRIPTION
**Pull request by martabal**
## Changes made in this PR

When the user is typing a person's name, Immich now suggests people to merge with based on the name entered.

## Screenshots

https://github.com/immich-app/immich/assets/74269598/2588b940-cc84-4347-a52a-3ff00e98f6fa

### CODE REVIEW

File: web/src/routes/(user)/people/[personId]/+page.svelte [hunk 1]
```diff
@@ -288,39 +329,65 @@
...
+        </section>
+        {#if isEditingName}
+          <div class="absolute z-[999] w-96">
+            {#each suggestedPeople as person, index (person.id)}
+              <div
+                class="flex {index === suggestedPeople.length - 1
+                  ? 'rounded-b-lg'
+                  : 'border-b dark:border-immich-dark-gray'} place-items-center bg-gray-100 p-2 dark:bg-gray-700"
+              >
+                <button class="flex w-full place-items-center" on:click={() => suggestPeople(person)}>
```
Comment (jrasm91) [inline web/src/routes/(user)/people/[personId]/+page.svelte:375]: Can you follow the convention of using `handle*` when handling a click event? So `handleSuggestPeople` here.

File: web/src/routes/(user)/people/[personId]/+page.svelte [hunk 2]
```diff
@@ -57,12 +58,34 @@
...
+          // Limit the size of filteredArray to 5
+          if (suggestedPeople.length === 5) {
+            break;
+          }
+        }
+      }
+    } else {
+      suggestedPeople = [];
+    }
+  }
```
Comment (jrasm91) [inline web/src/routes/(user)/people/[personId]/+page.svelte:87]: Is there any reason you can't turn this into something like `$: suggestedPeople = !name? [] : people.filter(...).slice(5)`

Comment (martabal) [reply web/src/routes/(user)/people/[personId]/+page.svelte]: Not really, it was for readability during my tests

File: web/src/routes/(user)/people/[personId]/+page.svelte [hunk 3]
```diff
@@ -58,12 +59,24 @@
...
+  let suggestedPeople: PersonResponseDto[] = [];
+
   $: isAllArchive = Array.from($selectedAssets).every((asset) => asset.isArchived);
   $: isAllFavorite = Array.from($selectedAssets).every((asset) => asset.isFavorite);
 
+  $: {
+    suggestedPeople = !name
+      ? []
+      : people
+          .filter((person: PersonResponseDto) => person.name.startsWith(name) && person.id !== data.person.id)
```
Comment (alextran1502) [inline web/src/routes/(user)/people/[personId]/+page.svelte:76]: Let's convert the input name and names' candidate to lowercase for better suggestion support in the filter method.

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**